interfaces/devdoc/file_util_linux.md

```diff
@@ -0,0 +1,69 @@
+# file_util_linux
+
+## Overview
+
+Linux implementation of the file operation functions.
+
+## Exposed API 
+
+```c
+// Defines removed for space constraints
+
+typedef void *HANDLE;
+typedef void *LPSECURITY_ATTRIBUTES;
+
+MOCKABLE_FUNCTION(, HANDLE, CreateFileA, const char*, lpFileName, unsigned long, dwDesiredAccess, unsigned long, dwShareMode, LPSECURITY_ATTRIBUTES, lpSecurityAttributes, unsigned long, dwCreationDisposition, unsigned long, dwFlagsAndAttributes, HANDLE, hTemplateFile);
+MOCKABLE_FUNCTION(, bool, CloseHandle, HANDLE, handle_input);
+```
+
+### CreateFileA
+
+```c
+MOCKABLE_FUNCTION(, HANDLE, CreateFileA, const char*, lpFileName, unsigned long, dwDesiredAccess, unsigned long, dwShareMode, LPSECURITY_ATTRIBUTES, lpSecurityAttributes, unsigned long, dwCreationDisposition, unsigned long, dwFlagsAndAttributes, HANDLE, hTemplateFile);
+```
+
+`CreateFileA` implements the creating file function in Linux. Uses the `open` function in C for Linux.
+
+**SRS_FILE_UTIL_LINUX_09_001: [** If the `full_file_name` input is either empty or `NULL`, `CreateFileA` shall return an `INVALID_HANDLE_VALUE`. **]**
+
+**SRS_FILE_UTIL_LINUX_09_002: [** `CreateFileA` shall allocate memory for the file handle and on success return a non-`NULL` value.  **]**
```

Comment:
>`CreateFileA` shall allocate memory for the file handle and on success return a non-`NULL` value. [](http://example.com/codeflow?start=34&length=97)

These can be two separate requirements:

`CreateFileA` shall allocate memory for the file handle.
`CreateFileA` shall succeed and return a non-`NULL` value.